stdlib/public/core/Span/RawSpan.swift

```diff
@@ -0,0 +1,702 @@
+//===--- RawSpan.swift ----------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// A RawSpan represents a span of initialized memory
+// of unspecified type.
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 9999, *)
+@frozen
+public struct RawSpan: ~Escapable, Copyable, BitwiseCopyable {
+  @usableFromInline let _pointer: UnsafeRawPointer?
+
+  @usableFromInline @inline(__always)
+  var _start: UnsafeRawPointer { _pointer.unsafelyUnwrapped }
+
+  @usableFromInline let _count: Int
+
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @usableFromInline @inline(__always)
+  @lifetime(immortal)
+  init(
+    _unchecked pointer: UnsafeRawPointer?,
+    byteCount: Int
+  ) {
+    _pointer = pointer
+    _count = byteCount
+  }
+}
+
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 9999, *)
+extension RawSpan: @unchecked Sendable {}
+
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 9999, *)
+extension RawSpan {
+
+  /// Unsafely create a `RawSpan` over initialized memory.
+  ///
+  /// The memory in `buffer` must be owned by the instance `owner`,
+  /// meaning that as long as `owner` is alive the memory will remain valid.
+  ///
+  /// - Parameters:
+  ///   - buffer: an `UnsafeRawBufferPointer` to initialized memory.
+  ///   - owner: a binding whose lifetime must exceed that of
+  ///            the newly created `RawSpan`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  @lifetime(immortal)
+  public init(
+    _unsafeBytes buffer: UnsafeRawBufferPointer
+  ) {
+    self.init(
+      _unchecked: buffer.baseAddress, byteCount: buffer.count
+    )
+  }
+
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  @lifetime(immortal)
+  public init(
```

Comment:
I'm asking Doug if this needs to be explicitly @unsafe. The strict memory safety vision isn't clear on whether generics parametrized with unsafe types are implicitly unsafe.

Reply:
Doug says this is implicitly unsafe, no change needed.

Reply:
Yeah, it's implicitly unsafe in the model that's experimentally implemented now. When `@unsafe` comes for a proper pitch and review, we'll have to decide what to do---this (and many other APIs) might end up being marked `@unsafe`, but we would do it as a group.